apps/frontend/src/pages/CreateAccount.tsx

```diff
@@ -1,5 +1,172 @@
+import React, { useState } from "react";
+import {
+  Box,
+  Button,
+  TextField,
+  Typography,
+  FormHelperText,
+  Link,
+} from "@mui/material";
+import martletPlaceLogo from "../images/martletplace-logo.png";
+import { colors } from "../styles/colors";
+import { useStyles } from "../styles/pageStyles";
+import { useNavigate } from "react-router-dom";
+import axios from "axios";
+
 const CreateAccount = () => {
-  return <h1>CreateAccount</h1>;
+  const classes = useStyles();
+  const [email, setEmail] = useState("");
+  const [username, setUsername] = useState("");
+  const [password, setPassword] = useState("");
+  const [passwordError, setPasswordError] = useState("");
+  const [emailError, setEmailError] = useState("");
+  const isFormEmpty = !username || !password || !email;
+  const navigate = useNavigate();
+
+  const handleCreateAccount = async (e: React.FormEvent) => {
+    e.preventDefault();
+
+    // Regex for password validation
+    const passwordFormat =
+      /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[\W_])[A-Za-z\d\W_]{8,}$/;
+
+    // Regex for UVic email validation
+    const emailFormat = /^[^@]+@uvic\.ca$/;
+
```

Comment:
@Scott-Kenning, are there any constraints on the length or types of characters in the username?

Reply:
We hadn't talked about it but it should probably be letters only for simplicity, maybe max 20 characters?